tests/import_proxy_test.go

```diff
@@ -0,0 +1,319 @@
+	. "github.com/onsi/ginkgo/extensions/table"
+	. "github.com/onsi/gomega"
+	"github.com/onsi/gomega/types"
+
+	configclient "github.com/openshift/client-go/config/clientset/versioned"
+	corev1 "k8s.io/api/core/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"kubevirt.io/containerized-data-importer/pkg/common"
+	"kubevirt.io/containerized-data-importer/pkg/controller"
+	"kubevirt.io/containerized-data-importer/tests/framework"
+	"kubevirt.io/containerized-data-importer/tests/utils"
+
+	cdiv1 "kubevirt.io/containerized-data-importer/pkg/apis/core/v1beta1"
+)
+
+const (
+	username         = "foo"
+	password         = "bar"
+	httpPort         = "8080"
+	httpPortWithAuth = "8081"
+	proxyServerName  = "cdi-test-proxy"
+	fileHostName     = "cdi-file-host"
+	//we use the a large image to "slowdown" the download process to have time to inspect the importer pod
+	cirrosFileName = "cirros-large-virtual-size.qcow2"
+)
+
+var _ = FDescribe("Import Proxy tests", func() {
+	type importProxyTestArguments struct {
+		name          string
+		size          string
+		url           func(bool) string
+		proxyURL      func(bool, bool) string
+		noProxy       string
+		isHTTPS       bool
+		withBasicAuth bool
+		dvFunc        func(string, string, string) *cdiv1.DataVolume
+		expected      func() types.GomegaMatcher
+	}
+	f := framework.NewFramework("import-proxy-func-test")
+
+	FDescribeTable("should", func(args importProxyTestArguments) {
+		var proxyHTTPURL string
+		var proxyHTTPSURL string
+		if args.isHTTPS {
+			proxyHTTPSURL = createProxyURL(args.isHTTPS, args.withBasicAuth, f.CdiInstallNs)
+		} else {
+			proxyHTTPURL = createProxyURL(args.isHTTPS, args.withBasicAuth, f.CdiInstallNs)
+		}
+		noProxy := args.noProxy
+		imgURL := createCirrosQcowURL(args.isHTTPS, f.CdiInstallNs)
+
+		By("Updating CDIConfig with ImportProxy configuration")
+		updateCDIConfigImporterProxyConfig(f, proxyHTTPURL, proxyHTTPSURL, noProxy)
+
+		By(fmt.Sprintf("Creating new datavolume %s", args.name))
+		dv := createHTTPDataVolume(f, args.name, args.size, imgURL, args.isHTTPS, args.withBasicAuth)
+		dataVolume, err := utils.CreateDataVolumeFromDefinition(f.CdiClient, f.Namespace.Name, dv)
+		Expect(err).ToNot(HaveOccurred())
+
+		By("Verifying pvc was created")
+		pvc, err := utils.WaitForPVC(f.K8sClient, dataVolume.Namespace, dataVolume.Name)
+		Expect(err).ToNot(HaveOccurred())
+		f.ForceBindIfWaitForFirstConsumer(pvc)
+
+		By("Verifying a message was printed to indicate a request was done using proxy")
+		verifyImporterPodInfoInProxyLogs(f, dataVolume, imgURL, args.isHTTPS, args.expected)
+
+		By("Waiting for the img import to be completed")
+		err = utils.WaitForDataVolumePhase(f.CdiClient, f.Namespace.Name, cdiv1.Succeeded, dataVolume.Name)
+		Expect(err).ToNot(HaveOccurred(), "Datavolume not in phase succeeded in time")
+
+		By("Deleting DataVolume")
+		err = utils.DeleteDataVolume(f.CdiClient, f.Namespace.Name, args.name)
```

Comment:
You delete the DV at the end of the test however, if something failed before that, the datavolume is not actually removed because that code is not executed. The normal way you do cleanup is in an:
```go
AfterEach(func() {
// Cleanup code goes here
})
```
Any setup that needs to happen goes in a "BeforeEach"